tests/consensus_tests/test_shards_consistency.py

```diff
@@ -0,0 +1,76 @@
+import multiprocessing
+import pathlib
+import random
+from datetime import datetime
+
+from .fixtures import create_collection, upsert_random_points
+from .utils import *
+
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger()
+
+N_PEERS = 3
+N_SHARDS = 1
+N_REPLICAS = 3
+COLLECTION_NAME = "test_collection"
+
+
+def update_points_in_loop(peer_url, collection_name):
+    limit = 5
+    while True:
+        offset = random.randint(0, 100)
+        upsert_random_points(peer_url, limit, collection_name, offset=offset, wait='false')
+
+def run_update_points_in_background(peer_url, collection_name):
+    p = multiprocessing.Process(target=update_points_in_loop, args=(peer_url, collection_name))
+    p.start()
+    return p
+
+
+def get_all_points(peer_url, collection_name):
+    res = requests.post(
+        f"{peer_url}/collections/{collection_name}/points/scroll",
+        json={
+            "limit": 100,
+            "with_vector": True,
+            "with_payload": True,
+        },
+        timeout=10
+    )
+    assert_http_ok(res)
+    return res.json()["result"]
+
+
+def test_shard_consistency(tmp_path: pathlib.Path):
+    assert_project_root()
+
+    peer_api_uris, peer_dirs, bootstrap_uri = start_cluster(tmp_path, N_PEERS)
+
+    create_collection(peer_api_uris[0], shard_number=N_SHARDS, replication_factor=N_REPLICAS)
+    wait_collection_exists_and_active_on_all_peers(collection_name="test_collection", peer_api_uris=peer_api_uris)
+
+    # upload points to the leader
+    upload_processes = [
+        run_update_points_in_background(peer_api_uris[i], "test_collection")
+        for i in range(len(peer_api_uris))
+    ]
+
+    print("Waiting for 5 seconds")
+    time.sleep(5)
+
+    # Kill all upload processes
+    for p in upload_processes:
+        p.kill()
```

Comment:
Points replication does not happen instantly. We should add a small delay before we start querying points, to let them propagate on all shards.

E.g., if we start querying right after killing uploaders:
- when we are querying first node, points 95-99 landed on the first node, but haven't been replicated on the other two
- when we are querying second node (just a moment later), points 95 and 96 landed on all nodes, but 97-99 are still being replicated
- and when we are querying the third node (another moment later), all points finally replicated everywhere
- and so each node will gather a slightly different set of responses
- and so, even with the read consistency feature (#1371, #1402), each node will resolve a slightly different result

With the pause added, the test (as is, without read consistency feature) will produce a false negative (i.e., the test will pass, even though Qdrant don't have any consistency features enabled).

However, running the test a few times will usually cause the test to fail (as it should) at least once.

```suggestion
    # Kill all upload processes
    for p in upload_processes:
        p.kill()

    print("Waiting for 5 seconds")
    time.sleep(5)
```

Reply:
Thank you for the analysis :+1: